fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -194,19 +194,45 @@ def reset_all
         all.each(&:reset)
       end
 
-      def copy_logarchive(device, dest)
+      def copy_logs(device, log_identity, dest)
+        dest = File.expand_path(dest)
+        os_version = ''
+        Open3.popen3("sw_vers -productVersion") do |stdin, stdout, stderr, wait_thr|
+          os_version = stdout.read.strip
+        end
+        if Gem::Version.new(os_version) >= Gem::Version.new('10.12.0')
+          copy_logarchive(device, log_identity, dest)
+        else
+          copy_logfile(device, log_identity, dest)
+        end
+      end
+
+      def copy_logfile(device, log_identity, dest)
+        logfile_src = File.expand_path("~/Library/Logs/CoreSimulator/#{device.udid}/system.log")
+        return unless File.exist?(logfile_src)
+
+        logfile_dst = File.join(dest, "system-#{log_identity}.log")
+
+        FileUtils.remove_file(logfile_dst, force: true)
+        FileUtils.cp(logfile_src, logfile_dst)
+        UI.success "Copying file '#{logfile_src}' to '#{logfile_dst}'..."
+      end
+
+      def copy_logarchive(device, log_identity, dest)
         sim_resource_dir = FastlaneCore::CommandExecutor.execute(command: "xcrun simctl getenv #{device.udid} SIMULATOR_SHARED_RESOURCES_DIRECTORY 2>/dev/null", print_all: false, print_command: true)
         logarchive_src = File.join(sim_resource_dir, "system_logs.logarchive")
+        logarchive_dst = File.join(dest, "system_logs-#{log_identity}.logarchive")
 
         # if logarchive already exists it fails as the .logarchive is a directory, so delete it. to be sure its gone
         FileUtils.rm_rf(logarchive_src)
-        FileUtils.rm_rf(dest)
+        FileUtils.rm_rf(logarchive_dst)
 
         command = "xcrun simctl spawn #{device.udid} log collect 2>/dev/null"
         FastlaneCore::CommandExecutor.execute(command: command, print_all: false, print_command: true)
 
-        FileUtils.cp_r(logarchive_src, dest)
-        UI.success "Copying file '#{logarchive_src}' to '#{dest}'..."
+        FileUtils.mkdir_p(logarchive_dst)
+        FileUtils.cp_r("#{logarchive_src}/.", logarchive_dst)
```

Comment:
This copies the _contents_ into the logarchive bundle which is a directory. Otherwise we are copying the src directory `blah.logarchive` into another directory named `specialblah.logarchive` => `blah.logarchive/specialblah.logarchive`